suite/src/benchmark.rs

```diff
@@ -20,56 +27,60 @@ pub fn get_cycle() -> u64 {
     ((counter_hi as u64) << 32u64) + counter_lo as u64
 }
 
+// TODO: implement general benchmarking function for sha265 module
+
 pub mod examples {
+
     use crate::platform::{self, Platform};
 
-    pub fn sha256_benchmark() {
+    /// Runs an example benchmark for the SHA256 module
+    ///
+    /// # Arguments
+    /// * `n` - the number of times the benchmark should be run
+    pub fn sha256_benchmark(n: u32) {
         if let Some(hmac_module) = platform::current().get_sha256_module() {
-            let input = [
-                0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0,
-                0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c,
-                0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc,
-                0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748,
-                0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb, 0x4057192d,
-                0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198, 0x04a92fdb,
-                0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119, 0xdf3f6198,
-                0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0, 0x14b81119,
-                0xdf3f6198, 0x04a92fdb, 0x4057192d, 0xc43dd748, 0xea778adc, 0x52bc498c, 0xe80524c0,
-                0x14b81119,
-            ];
-            let mut output = [0u32; 8];
+            println!("initialization, computation, reading_output");
+
+            for _ in 0..n {
+                let input = [
```

Comment:
Can you define the input outside the for loop, as it only needs to be defined once?